gpytorch/module.py

```diff
@@ -262,6 +262,20 @@ def setting_closure(module, val):
                 )
             closure = param_or_closure
 
+        if prior is not None:
+            hyperparameter_shape = closure(self).shape
+            prior_shape = prior.shape()
```

Comment:
Did this work at any point? Is this meant to be `event_shape`?

Alternatively we would give `Prior`s a `shape` property that is just `self._batch_shape + self._event_shape`.